This patch is a fix for a bug in st40r2-4.0.1 ONLY.
It should not need to be a applied to any other version.

--- src/gcc/config/sh/sh.h.orig	2006-08-11 10:48:30.000000000 +0100
+++ src/gcc/config/sh/sh.h	2006-08-11 10:48:44.000000000 +0100
@@ -481,11 +481,6 @@
       target_flags |= MASK_SMALLCODE;					\
       sh_div_str = SH_DIV_STR_FOR_SIZE ;				\
     }									\
-  else									\
-    {									\
-      TARGET_CBRANCHDI4 = 1;						\
-      TARGET_EXPAND_CBRANCHDI4 = 1;					\
-    }									\
   /* We can't meaningfully test TARGET_SHMEDIA here, because -m options	\
      haven't been parsed yet, hence we'd read only the default.		\
      sh_target_reg_class will return NO_REGS if this is not SHMEDIA, so	\
